### DISCUSSION TITLE
custom validation after submit

### DISCUSSION THREAD

**Discussion by novaknole** [react-hook-form/react-hook-form#5509] [General]
Hi @bluebill1049 @thanh-nguyen-95

I have 2 text input fields and `submit` button as well.

I need those 2 fields to have some validation functions, but I don't want to have `validate` on them. 

When I click `submit`,  which calls `save` function, that's where I want to do my own validation on those 2 input fields and if something is wrong, I want to stop and show the user the actual error, but I want this error to be shown as the error part of the form validation. 1. This would stop the execution 2.  it would put the message in the `errors` which would appear on below of the field that the validation failed for.

Is it possible ?

**Comment by thanh-nguyen-95** [comment]
It's possible using `setError`. An example: https://codesandbox.io/s/quizzical-poincare-c1crf?file=/src/App.js
